components/switch/xiaomi.py

```diff
@@ -78,38 +55,20 @@ def turn_on(self, **kwargs):
         """Turn the switch on."""
         if self.xiaomi_hub.write_to_hub(self._sid, self._data_key, 'on'):
             self._state = True
-            self.schedule_update_ha_state()
 
     def turn_off(self):
         """Turn the switch off."""
         if self.xiaomi_hub.write_to_hub(self._sid, self._data_key, 'off'):
             self._state = False
-            self.schedule_update_ha_state()
-
-    def toggle(self):
-        if self._state == False:
-            self.turn_on()
-        else:
-            self.turn_off()
 
     def parse_data(self, data):
-        if self._data_key not in data:
+        value = data.get(self._data_key)
+        if value is None:
             return False
 
-        state = True if data[self._data_key] == 'on' else False
+        state = value == 'on'
         if self._state == state:
             return False
         else:
             self._state = state
             return True
-
-    def push_data(self, data):
-        """Push from Hub"""
-        if self.parse_data(data):
-            self.schedule_update_ha_state()
-
-    def update(self):

```

Comment:
The push_function is now part of the XiaomiDevice since it is generic for all the devices: https://github.com/lazcad/homeassistant/blob/master/components/xiaomi.py#L374